src/type_translator.ts

```diff
@@ -658,7 +658,18 @@ export class TypeTranslator {
         typeArgs = typeArgs.slice(0, typeArgs.length - 1);
       }
       if (typeArgs.length > 0) {
-        const params = typeArgs.map(t => this.translate(t));
+        const params = typeArgs.map(t => {
+          // If a type references itself recursively, such as in `type A =
```

Comment:
Do you know why L423 doesn't help here?  Should it be broadened to just "seenTypes" perhaps?

Also, has this always been broken or is it new?